# Семинар 4 ДЗ

## Задание 1
Случайная непрерывная величина $A$ имеет равномерное распределение на промежутке $(200, 800]$.

Найдите ее среднее значение и дисперсию.

Используем формулы:

$M(x)=\dfrac {a+b}{2}$

$D= \dfrac{(b-a)^2}{12}$

$M(x)=\dfrac {200+800}{2} = \dfrac {1000}{2} = 500$

$D= \dfrac{(800-200)^2}{12} = \dfrac{(800-200)^2}{12} = \dfrac{(600)^2}{12} = 30000$



## Задание 2
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.

Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?

Если да, найдите ее.

Используем формулу:

$D= \dfrac{(b-a)^2}{12}$

Подставим значения:

$0.2=\dfrac{(b-0.5)^2}{12}$

$(b-0.5)^2 = 2.4$, возведем скобку в квадрат

$b^2 - b + 0.25 - 2.4 = 0$

$b^2 - b - 2.15 = 0$

Найдем корни этого квадратного уравнения:

In [2]:
import numpy as np

def solve_quadratic_equation(a, b, c):
    discriminant = b**2 - 4*a*c
    if discriminant > 0:
        x1 = (-b + np.sqrt(discriminant)) / (2*a)
        x2 = (-b - np.sqrt(discriminant)) / (2*a)
        return x1, x2
    elif discriminant == 0:
        x1 = -b / (2*a)
        return x1
    else:
        return None

# Задаем коэффициенты уравнения
a = 1
b = -1
c = -2.15

# Решение уравнения и вывод результатов
result = solve_quadratic_equation(a, b, c)
if result is None:
    print("Уравнение не имеет действительных корней")
elif isinstance(result, tuple):
    print(f"Корни уравнения: x1 = {result[0]}, x2 = {result[1]}")
else:
    print(f"Корень уравнения: x = {result}")

Корни уравнения: x1 = 2.049193338482967, x2 = -1.0491933384829668


Так как нам известно, что $0.5$ это левая граница, то нам подходит корень $\approx 2.05$, следовательно

$M(x)=\dfrac {a+b}{2}=\dfrac {0.5+2.05}{2} = 1.275$


## Задание 3
Непрерывная случайная величина X распределена нормально и задана плотностью распределения

$f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)$

Найдите:

1. $M(X)$

2. $D(X)$

3. $std(X)$ (среднее квадратичное отклонение)

$f(x) = \dfrac{1}{4 \cdot \sqrt{2\pi}} \cdot e^{-\dfrac{(x+2)^2}{32}}$

Используем формулу плотности распределения

$f(x) = \dfrac{1}{σ \sqrt {2 \pi}} e^{-\dfrac{(x-\mu)^2}{2σ^2}}$

и сравним значения:

Среднее арифметическое $M(X) = \mu = -2$

Дисперсия $D(X) = \sigma ^2 = 32 / 2 = 16$

$std(X)$ (среднее квадратичное отклонение) $\sigma = \sqrt{16} = 4$


## Задание 4
Рост взрослого населения города $X$ имеет нормальное распределение.

Причем, средний рост равен $174$ см, а среднее квадратичное отклонение равно $8$ см.

Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:

а). больше 182 см

б). больше 190 см

в). от 166 см до 190 см

г). от 166 см до 182 см

д). от 158 см до 190 см

е). не выше 150 см или не ниже 190 см

ё). не выше 150 см или не ниже 198 см

ж). ниже 166 см.

Данную задачу можно решить двумя способами:

### Способ 1

Если внимательно посмотреть на искомые интервалы, то можем обратить внимание на возможность применения правила $3\sigma$ трёх сигм

![Три сигма](https://upload.wikimedia.org/wikipedia/commons/thumb/3/37/Standard_deviation_diagram_%28decimal_comma%29.svg/325px-Standard_deviation_diagram_%28decimal_comma%29.svg.png)


$-3\sigma = 174 - 3 * 8 = 150$<br>
$-2\sigma = 174 - 2 * 8 = 158$<br>
$-\sigma = 174 - 8 = 166$<br>
$\sigma = 174 + 8 = 182$<br>
$2\sigma = 174 + 2 * 8 = 190$<br>
$3\sigma = 174 + 3 * 8 = 198$<br>





In [4]:
from scipy import stats

а). больше $182$ см

больше $\sigma = 13.6\% + 2.1\% + 0.1\% = 15.8\%$

или

$100\% - 50\% - 34.1\% = 15.9\%$

Проверим $2$ способом, при помощи python

In [5]:
# а). больше 182 см
1 - stats.norm.cdf(182, loc=174, scale=8)

0.15865525393145707

б). больше $190$ см

больше $2\sigma = 2.1\% + 0.1\% = 2.2\%$

In [6]:
# б). больше 190 см
1 - stats.norm.cdf(190, loc=174, scale=8)

0.02275013194817921

в). от $166$ см до $190$ см

от $−σ$ до $+2σ = 68.2\% + 13.6\% = 81.8\%$

In [7]:
# в). от 166 см до 190 см

stats.norm.cdf(190, loc=174, scale=8) - stats.norm.cdf(166, loc=174, scale=8)

0.8185946141203637

г). от $166$ см до $182$ см

от $−σ$ до $+σ$ $=68.2\%$

In [8]:
# г). от 166 см до 182 см

stats.norm.cdf(182, loc=174, scale=8) - stats.norm.cdf(166, loc=174, scale=8)

0.6826894921370859

д). от $158$ см до $190$ см

от $−2σ$ до $+2σ = 95.4\%$

In [9]:
# д). от 158 см до 190 см

stats.norm.cdf(190, loc=174, scale=8) - stats.norm.cdf(158, loc=174, scale=8)

0.9544997361036416

е). не выше $150$ см или не ниже $190$ см

меньше $−3σ$ и больше $2σ = 0.1\% + 2.1\% + 0.1\% = 2.3\%$

In [10]:
# е). не выше 150 см или не ниже 190 см

1 - stats.norm.cdf(190, loc=174, scale=8) + stats.norm.cdf(150, loc=174, scale=8)

0.0241000299798093

ё). не выше $150$ см или не ниже $198$ см

меньше $−3σ$ и больше $3σ = 0.1\% + 0.1\% = 0.2\%$

In [11]:
# ё). не выше 150 см или не ниже 198 см

1 - stats.norm.cdf(198, loc=174, scale=8) + stats.norm.cdf(150, loc=174, scale=8)

0.0026997960632601965

ж). ниже $166$ см.

меньше $−σ = 13.6\% + 2.1\% + 0.1\% = 15.8\%$ 

In [12]:
# ж). ниже 166 см.
stats.norm.cdf(166, loc=174, scale=8)

0.15865525393145707


## Задание 5
На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

Применим формулу нормирования:

$Z=\dfrac {X-μ}{σ}$

$Z=\dfrac {190-178}{\sqrt{25}} = \dfrac{12}{5} = 2.4\sigma$